src/distributions/utils.rs

```diff
@@ -162,3 +162,195 @@ macro_rules! simd_less_then {
 #[cfg(feature="simd_support")] simd_less_then! { f64x2 }
 #[cfg(feature="simd_support")] simd_less_then! { f64x4 }
 #[cfg(feature="simd_support")] simd_less_then! { f64x8 }
+
+
+// Until portable shuffles land in stdsimd, we expose and use the shuffle intrinsics directly.
+#[cfg(feature = "simd_support")]
+extern "platform-intrinsic" {
+    fn simd_shuffle2<T, U>(a: T, b: T, indices: [u32; 2]) -> U;
+    fn simd_shuffle4<T, U>(a: T, b: T, indices: [u32; 4]) -> U;
```

Comment:
Where's the documentation on these?

Reply:
I don’t think there is any. I think this is where they’re defined https://github.com/rust-lang/rust/blob/2612bbcba08dd81730edd8f2139005fb7a409294/src/librustc_codegen_llvm/intrinsic.rs#L1056